src/components/DeleteButton.spec.js

```diff
@@ -0,0 +1,29 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import DeleteButton from './DeleteButton.js';
+
+describe('DeleteButton', () => {
+  it('renders a button, svg and sr-only text', () => {
+    const deleteClick = jest.fn();
+    render(<DeleteButton onClick={deleteClick} />);
+    const button = screen.getByRole('button', { name: /Delete/i });
+
+    const srOnlyText = screen.getByText(/Delete/i);
+
+    const svg = screen.getByTestId("delete-icon")
+    expect(svg).toBeInTheDocument()
+
+    expect(button).toBeInTheDocument();
+    expect(srOnlyText).toBeInTheDocument();
+  });
+
+  it('renders clicking the button', () => {
```

Comment:
maybe rewrite the it Statement a little.
What you want to test is, if the function is called by clicking the button. So maybe something like this:
'calls the mock function when the button is clicked'